tests/cephadm/test_custom_container.py

```diff
@@ -0,0 +1,162 @@
+
+
+def run(ceph_cluster, **kw):
+    """Verify re-deploying of monitoring stack with custom images
+    Args:
+        ceph_cluster(ceph.ceph.Ceph): CephNode or list of CephNode object
+    """
+
+    # Get config file and node
+    config = kw.get("config")
+
+    # Get registry credentials
+    registry, reg_username, reg_password = get_registry_details(config)
+
+    # Get installer node
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Define CephAdm class object
+    cephadm = CephAdm(installer)
+
+    # Get cluster nodes
+    ceph_nodes = kw.get("ceph_nodes")
+
+    # Get dashboard image stack
+    registry_image = config.get("dashboard_images")
+    if not registry_image:
+        raise Exception("Dashboard images are not provided")
+
+    # Get dashboard images
+    dashboard_images = registry_image.values()
+
+    # Get registry docker image
+    docker_reg_image = config.get("registry_image").get("docker_registry_image")
+
+    # Get the rhbuild
+    rhbuild = config.get("rhbuild")
+    if rhbuild.startswith("6"):
+        rhbuild = "quincy"
+    elif rhbuild.startswith("5"):
+        rhbuild = "pacific"
+    else:
+        raise Exception("Unsupported version")
+
+    # Setup ssh key to all nodes
+    setup_ssh_keys(installer, ceph_nodes)
+
+    # Setup a private registry on installer node
+    setup_private_container_registry(
+        installer=installer,
+        nodes=ceph_nodes,
+        registry=registry,
+        reg_username=reg_username,
+        reg_password=reg_password,
+        build_type=rhbuild,
+        private_reg_username=PRIVATE_REG_USER,
+        private_reg_password=PRIVATE_REG_PASS,
+        docker_reg_image=docker_reg_image,
+        images=dashboard_images,
+    )
+
+    # Get private registry catalog
+    out = get_private_registry_image(
+        installer, username=PRIVATE_REG_USER, password=PRIVATE_REG_PASS
+    )
+    images = out.get("repositories")
+
+    # Pull images to nodes
+    for ceph in ceph_nodes:
+        Registry(ceph).login(
+            registry=f"{installer.hostname}:{PORT}",
+            username=PRIVATE_REG_USER,
+            password=PRIVATE_REG_PASS,
+        )
+        for img in images:
+            Container(ceph).pull(image=f"{installer.hostname}:{PORT}/{img}")
+
+    for config_key, image in registry_image.items():
+        img_reg = image.replace(f"{registry}/", "")
+
+        # Set mgr config for dashboard images
+        conf = {
+            "key": "mgr",
+            "value": f"mgr/cephadm/{config_key} {installer.hostname}:{PORT}/{img_reg}",
+        }
+        cephadm.ceph.config.set(**conf)
+
+        # Get mgr config for dashboard images
+        conf = {
+            "who": "mgr",
+            "key": f"mgr/cephadm/{config_key}",
+        }
+        out = cephadm.ceph.config.get(**conf)[0]
+
+        # Validate if dashboard images are updated correctly
+        if installer.hostname not in out:
+            raise ClusterCustomContainerError(
```

Comment:
instead use - https://github.com/red-hat-storage/cephci/blob/master/cli/exceptions.py#L32

Reply:
Okay, updated the changes.